nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -66,24 +70,33 @@ public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
             // Write to a temporary output in case of exception during write()
             tb.writeStartObject();
             write(tb, enrichable.getEntity());
+            tb.writeEndObject();
             tb.flush();
             // Add the complete, well-formed content to the real output
             try (JsonParser parser = tb.asParser()) {
                 parser.nextToken(); // ignoring START_OBJECT
-                while (parser.nextToken() == JsonToken.FIELD_NAME) {
+                while (parser.nextToken() == FIELD_NAME) {
                     jg.copyCurrentStructure(parser);
                 }
-                if (parser.currentToken() != null) {
-                    log.error("Enricher {} returned invalid output {}", name, tb.toString());
+                Supplier<String> supplier = () -> {
+                    try {
+                        return MAPPER.readTree(tb.asParser());
+                    } catch (IOException e) {
+                        return "malformed content could not be retrieved";
+                    }
+                };
+                if (parser.currentToken() != END_OBJECT) {
+                    log.error("Enricher: {} failed on current token: {}, output to write: {}", name::toString,
+                            parser::currentToken, supplier::get);
```

Comment:
Could you extract the Supplier as a protected method? You'll be able to call it as a lambda with (note _this_ is optional):
```
() -> this.safeReadBuffer(tb)
```
I prefer a method here because it's more readable and have the same number of line than the supplier declaration.

IMHO, one valid Supplier as variable case is when:
- supplier doesn't use curly brackets
- the lambda makes the code format ugly

Reply:
I did it but removed the "this." part as we are calling it in the context of a lambda